### DISCUSSION TITLE
Storage: Byte range requests?

### DISCUSSION THREAD

**Discussion by quicksnap** [supabase/supabase#1558] [Questions] [answered]
I'm trying to leverage a Storage bucket with pre-signed URLs to play back audio on iOS. When trying to trigger the sound, I get 

```
[Error: The server is not correctly configured. - The AVPlayerItem instance has failed with the error code -11850 and domain "AVFoundationErrorDomain".]
```

When I use [the following remote test file](https://dl.espressif.com/dl/audio/gs-16b-2c-44100hz.m4a) (NOT in the bucket) it works fine. After digging on Google for a bit, I am pretty sure this error is due to the storage server not supporting byte range requests. The only relevant difference between the remote URL and the bucket response headers is the `Accept-ranges` header.

To my knowledge, S3 signed urls support byte range requests, and I _think_ you all are using S3 under the hood?

So, are byte range requests for pre-signed URLs possible? 

For now, I'll probably just not stream it and just download-and-play, since I'm still prototyping. Eventually, though, having a byte range request would be great. Maybe I could contribute something to the code here if it's feasible!

**Comment by inian** [comment, marked as answer]
Hey @quicksnap, I haven't checked this but yeah we should support this if we aren't already. Creating an issue [here](https://github.com/supabase/storage-api/issues/20)

[Here](https://github.com/supabase/storage-api/blob/ed66a533b58ece4f73fb54ca051de90a351094d1/src/routes/object/getSignedObject.ts#L61) is where we stream the response from s3. So I think we will need to pass through the headers from s3 to the client there. Let me know if want to contribute to the code and I can help you get up and running if you are stuck somewhere :)

**Reply by quicksnap** [reply]
Awesome! Thanks for the quick reply.

Looks like the work itself may be simple! For the short-term, I need to keep working on my prototype. If I get some time or get blocked, I'll set up a supabase dev env and help out.

Thanks for the awesome stuff so far. It's turbo-charged my work, for sure. Incredible.
